### ISSUE TITLE
mysql stats stop working at first update

### ISSUE THREAD

**Issue by bbbart** [mikaku/Monitorix#214]
Hi,

I have a monitorix monitoring a bunch of stuff on my server, but I can't seem to get the MySQL module to work.

I configured it as follows:

    <mysql>
         conn_type = socket
         list = /var/run/mysqld/mysqld.sock
         <desc>
                 /var/run/mysqld/mysqld.sock = 3306,monitorix,<XXX>
         </desc>
         rigid = 0, 2, 0, 0, 0, 0
         limit = 100, 100, 100, 100, 100, 100
    </mysql>

and in the logfile I see no errors related to this configuration. Moreover, I even see correct values in the MySQL portion of the monitorix graphs.

However, after that first initial successful connection, monitorix seems to fail to reconnect:

    Thu Sep 27 11:08:50 2018 - Starting Monitorix version 3.10.1 (pid 27548).
    Thu Sep 27 11:08:50 2018 - Loaded main configuration file '/etc/monitorix/monitorix.conf'.
    Thu Sep 27 11:08:50 2018 - Loading extra configuration file '/etc/monitorix/conf.d/00-debian.conf'.
    Thu Sep 27 11:08:50 2018 - Initializing graphs.
    Thu Sep 27 11:08:50 2018 - Generating the 'index.html' file.
    Thu Sep 27 11:08:50 2018 - Started built-in HTTP server (pid 27578).
    Thu Sep 27 11:08:50 2018 - Ok, ready.
    Thu Sep 27 11:09:02 2018 - mysql::mysql_update: Cannot connect to MySQL '/var/run/mysqld/mysqld.sock'.
    Thu Sep 27 11:09:02 2018 - ERROR: while updating /var/lib/monitorix/mysql.rrd: /var/lib/monitorix/mysql.rrd: expected timestamp not found in data source from N
    HTTPServer: You can connect to your server at http://localhost:9556/
    Thu Sep 27 11:10:02 2018 - mysql::mysql_update: Cannot connect to MySQL '/var/run/mysqld/mysqld.sock'.
    Thu Sep 27 11:10:02 2018 - ERROR: while updating /var/lib/monitorix/mysql.rrd: /var/lib/monitorix/mysql.rrd: expected timestamp not found in data source from N
    Thu Sep 27 11:11:01 2018 - mysql::mysql_update: Cannot connect to MySQL '/var/run/mysqld/mysqld.sock'.
    Thu Sep 27 11:11:01 2018 - ERROR: while updating /var/lib/monitorix/mysql.rrd: /var/lib/monitorix/mysql.rrd: expected timestamp not found in data source from N
    Thu Sep 27 11:12:02 2018 - mysql::mysql_update: Cannot connect to MySQL '/var/run/mysqld/mysqld.sock'.
    Thu Sep 27 11:12:02 2018 - ERROR: while updating /var/lib/monitorix/mysql.rrd: /var/lib/monitorix/mysql.rrd: expected timestamp not found in data source from N
    Thu Sep 27 11:13:02 2018 - mysql::mysql_update: Cannot connect to MySQL '/var/run/mysqld/mysqld.sock'.
    Thu Sep 27 11:13:02 2018 - ERROR: while updating /var/lib/monitorix/mysql.rrd: /var/lib/monitorix/mysql.rrd: expected timestamp not found in data source from N
    Thu Sep 27 11:14:02 2018 - mysql::mysql_update: Cannot connect to MySQL '/var/run/mysqld/mysqld.sock'.
    Thu Sep 27 11:14:02 2018 - ERROR: while updating /var/lib/monitorix/mysql.rrd: /var/lib/monitorix/mysql.rrd: expected timestamp not found in data source from N
    Thu Sep 27 11:15:01 2018 - mysql::mysql_update: Cannot connect to MySQL '/var/run/mysqld/mysqld.sock'.
    Thu Sep 27 11:15:01 2018 - ERROR: while updating /var/lib/monitorix/mysql.rrd: /var/lib/monitorix/mysql.rrd: expected timestamp not found in data source from N

resulting in a graph like this:
![mysql](https://user-images.githubusercontent.com/3244673/46136161-399f2e00-c247-11e8-8865-4d1024ada9e8.png)

Could someone point me to what might be going on?

This is on a Debian 9.0 system with monitorix 3.10.1 from the izzysoft repository and mariadb-server 10.1.

Thanks a lot!

**Comment by mikaku** [comment]
Hi,

This is a common issue. Look that line from your log file:

```
Thu Sep 27 11:08:50 2018 - Loading extra configuration file '/etc/monitorix/conf.d/00-debian.conf'.
```

:-)

and as stated in the Monitorix web site (https://www.monitorix.org/doc-debian.html):

IMPORTANT NOTICE:
--
The Debian package also comes with an extra configuration file in `/etc/monitorix/conf.d/00-debian.conf` that includes some options specially adapted for Debian systems. **This file will be loaded right after the main configuration file**, hence some options in the main configuration will be overwritten by this extra file.

So, either, remove or change the name of the extra configuration file, or put in that file your MySQL configurations.

I hope that helped you.

**Comment by bbbart** [comment]
Well, I got that far and had removed the MySQL section from the 00-debian.conf file already.

Strangely enough, I installed monitorix similarly on another server with the same setup, except for a different MySQL password for the monitorx user, and that worked just fine.

So, I changed the password for the monitorix MySQL user on the server I reported this issue for and BAM, all was well. The password I used before had a lot of 'strange' characters (yet, all ascii) in it, which might explain things?

Anyway, got a nice graph now. :-)
